Maple2.Server.Game/PacketHandlers/PartyHandler.cs

```diff
@@ -0,0 +1,249 @@
+                HandleCancelPartySearch(session, packet);
+                return;
+            case Command.VoteKick:
+                HandleVoteKick(session, packet);
+                return;
+            case Command.ReadyCheck:
+                HandleReadyCheck(session, packet);
+                return;
+            case Command.ReadyCheckResponse:
+                HandleReadyCheckResponse(session, packet);
+                return;
+        }
+
+    }
+
+    private void HandleInvite(GameSession session, IByteReader packet) {
+        if (session.Party.Party == null) {
+            // Create new party
+            PartyResponse response = World.Party(new PartyRequest {
+                RequestorId = session.CharacterId,
+                Create = new PartyRequest.Types.Create { },
+            });
+
+            var error = (PartyError) response.Error;
+            if (error != PartyError.none) {
+                session.Send(PartyPacket.Error(error));
+                return;
+            }
+
+            session.Party.SetParty(response.Party);
+        }
+
+        string playerName = packet.ReadUnicodeString();
+        using GameStorage.Request db = session.GameStorage.Context();
+        long characterId = db.GetCharacterId(playerName);
+        if (characterId == 0) {
+            session.Send(PartyPacket.Error(PartyError.s_party_err_cannot_invite));
+            return;
+        }
+
+        try {
+            var request = new PartyRequest {
+                RequestorId = session.CharacterId,
+                Invite = new PartyRequest.Types.Invite {
+                    PartyId = session.Party.Id,
+                    ReceiverId = characterId,
+                },
+            };
+            PartyResponse response = World.Party(request);
+            var error = (PartyError) response.Error;
+            if (error != PartyError.none) {
+                session.Send(PartyPacket.Error(error, playerName));
+                return;
+            }
+        } catch (RpcException ex) {
+            Logger.Error(ex, "Failed to invite {Name} to party", playerName);
+            session.Send(PartyPacket.Error(PartyError.s_party_err_not_found));
+        }
+    }
+
+    private void HandleInviteResponse(GameSession session, IByteReader packet) {
+        string name = packet.ReadUnicodeString();
+        PartyInvite.Response response = (PartyInvite.Response) packet.ReadByte();
+        int partyId = packet.ReadInt();
+
+        PartyResponse partyResponse = World.Party(new PartyRequest {
+            RequestorId = session.CharacterId,
+            RespondInvite = new PartyRequest.Types.RespondInvite {
+                PartyId = partyId,
+                Reply = (int) response,
+            },
+        });
+        var error = (PartyError) partyResponse.Error;
+        if (error != PartyError.none) {
+            session.Send(PartyPacket.Error(error));
+            return;
+        }
+
+        if (response == PartyInvite.Response.Accept) {
```

Comment:
We'll also need to check if the party still exists.

Reply:
take this scenario:
Player 1 invites Player 2 to their party.
Player 1 decides to disband (for whatever reason, logged out, their decision, joined a different party)
Player 2 still has the prompt to join Player 1's party


If Player 1 accepts, it should return that the party no longer exists.

Reply:
Player 1 would still have the prompt, but when Player 1 accepts, it will make a request to `WorldServer` to join the party. At this point, `WorldServer` knows the party is disbanded, so the request to join the party *should* fail and Player 1 will get an error instead.